OpenRA.Mods.D2k/Traits/Buildings/D2kBuilding.cs

```diff
@@ -0,0 +1,100 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits.Buildings
+{
+	public class D2kBuildingInfo : BuildingInfo
+	{
+		[Desc("Amount of damage received per DamageInterval ticks.")]
+		public readonly int Damage = 500;
+
+		[Desc("Delay between receiving damage.")]
+		public readonly int DamageInterval = 100;
+
+		[Desc("Apply the damage using these damagetypes.")]
+		public readonly BitSet<DamageType> DamageTypes = default(BitSet<DamageType>);
+
+		[Desc("Terrain types where the actor will take damage.")]
+		public readonly string[] DamageTerrainTypes = { "Rock" };
+
+		[Desc("Percentage health below which the actor will not receive further damage.")]
+		public readonly int DamageThreshold = 50;
+
+		[Desc("Inflict damage down to the DamageThreshold when the actor gets created on damaging terrain.")]
+		public readonly bool StartOnThreshold = true;
```

Comment:
AFAIK this isn't used / useful for the D2k building usecase, so could perhaps be removed?

Reply:
Yeah, I wondered about this too, but then thought this might be useful in the future if we want a building to start with full health but then take damage. I can remove it as practically dead code if you want.

Reply:
Ok, lets keep it then.